content/doc/book/using/working-with-projects.adoc

```diff
@@ -38,6 +44,12 @@ The existing project will be copied to a new project with the name that was ente
2. But if we go down into the script block, you can see that the job definition is here. So let's make a small change to this. I'll say hello world to, okay. Let's click on save. And now let's click on build now to make sure it actually works. And as it completes, we can see that it finished up and hello world to was output. And let's go back up to our dashboard, and we can see at the root of our Jenkins controller, we have two jobs, test dash pipeline and test dash pipeline dash two. Now, what if I wanted to do this from the command line? So let's go ahead and go over into Managed Jenkins, scroll down to the bottom and click on Jenkins CLI. If you've never downloaded the Jenkins CLI before, you can download it right here from the top of the page. And the command that we're gonna be wanting to run is Copy Job. And if you take a look at the syntax for Copy Job, we'll say Copy Job, we'll give it the source, and then we'll give it the destination. So the existing job and what the new job is. Let's go over to our command prompt, and let's set up a few environment variables. The first environment variable that we're gonna set up is Jenkins's user ID, and we're setting it to admin, because that's the user that I already have set up on this controller. The second environment variable that we're setting is Jenkins_API_token. And this is the token associated with the admin user. If you click on admin and go to configure, you'll see that there's already an API token here, and the value of this token is what we have in our shell. And finally, we're going to set up our Jenkins URL, which is the URL for our server, which is Jenkins' colon AD80. Now let's go ahead and run our copy job command. So we'll say java-jar, jenkins-cli.jar, copy job. We're going to copy from our original job, which is test-pipeline. And the new name is going to be test-pipeline-3. So when we hit enter, it's going to copy that job. But we don't see anything in the command prompt. So let's go back over to our dashboard. And you'll see now that we have test-pipeline-3. Now, unlike when we did the copy within the UI, where we were able to make changes before we clicked on Save. Here, it just copied from our source pipeline in this case over to our destination pipeline, which is test pipeline three. So I can click into this, click on Configure, make any extra changes that I want to make. In this case, I'm just going to add a three, click Save, and then click on Build Now. Now that we have test pipeline three, we've made the change, and now the output is Hello World Three. If you have any questions or comments, you can reach out to us on Twitter at CloudBees. If this video was helpful to you, give us a thumbs up. And if you haven't subscribed to CloudBeesTV yet, why not? Take a moment, click on that subscribe button, and then ring that bell, and you'll be notified anytime there's new content available on CloudBeesTV. Thanks for watching, and we will see you in the next video. (upbeat music)
```

Comment:
How to clone a Jenkins job.

Here's today's starting point: I have a Jenkins LTS controller version 2.346.3 and on this controller, I already have a sample job setup.
Now the sample job is just a very simple test Pipeline.
In fact, the name even says it.
If we take a look at the configuration, no options are checked and we have a basic Pipeline script already defined within the script block.
Let's go back up to dashboard and let's create a *new item*.
So we're wanting to clone "test-pipeline".
So in this case, I'm going to say "test-pipeline-2", but instead of picking Pipeline, I'm going to scroll to the bottom and I'm going to say copy from and I'm going to start typing here.
I'm going to say "T" and it automatically finds the "test-pipeline" for me.
So I'll select that and click on *OK*.

Now, you'll notice up in the breadcrumb, we now have "test-pipeline-2".
But if we go down into the script block, you can see that the job definition is here.
So let's make a small change to this, I'll say `'Hello World 2'` okay.
Let's click on save and now, let's click on build now to make sure it actually works.
As it completes, we can see that it finished up and `Hello World 2` was output.
Let's go back up to our dashboard, and we can see at the root of our Jenkins controller, we have two jobs, "test-pipeline" and "test-pipeline-2".
Now, what if I wanted to do this from the command line?

Let's go ahead and go over into Manage Jenkins, scroll down to the bottom and click on *Jenkins CLI*.
If you've never downloaded the Jenkins CLI before, you can download it right here from the top of the page.
The command that we're gonna be wanting to run is `copy-job`.
If you take a look at the syntax for `copy-job`, we'll say `copy-job`, we'll give it the source, and then we'll give it the destination.
So the existing job and what the new job is.
Let's go over to our command prompt, and let's set up a few environment variables.

The first environment variable that we're gonna set up is `JENKINS_USER_ID`, and we're setting it to `admin`, because that's the user that I already have set up on this controller.
The second environment variable that we're setting is `JENKINS_API_TOKEN`.
This is the token associated with the admin user.
If you click on admin and go to configure, you'll see that there's already an API token here, and the value of this token is what we have in our shell.
Finally, we're going to set up our `JENKINS_URL`, which is the URL for our server, which is `jenkins:8080`.
Now let's go ahead and run our copy job command.
So we'll say `java -jar jenkins-cli.jar copy job test-pipeline test-pipeline-3`.
We're going to copy from our original job, which is "test-pipeline".
The new name is going to be "test-pipeline-3".
So when we hit enter, it's going to copy that job, but we don't see anything in the command prompt.

Let's go back over to our dashboard and you'll see now that we have "test-pipeline-3".
Now, unlike when we did the copy within the UI, where we were able to make changes before we clicked on *Save*.
Here, it just copied from our source Pipeline in this case over to our destination Pipeline, which is "test-pipeline-3".
So I can click into this, click on *Configure*, and make any extra changes that I want to make.
In this case, I'm just going to add a three, click *Save*, and then click on *Build Now*.
Now that we have "test-pipeline-3", we've made the change, and now the output is `Hello World 3`.

If you have any questions or comments, you can reach out to us on Twitter @CloudBees.
If this video was helpful to you, give us a thumbs up.
And if you haven't subscribed to CloudBeesTV yet, why not?
Take a moment, click on that subscribe button, and then ring that bell, and you'll be notified anytime there's new content available on CloudBeesTV.
Thanks for watching, and we will see you in the next video.